### PR TITLE
GDB-5337 Mapping UI UX improvements

### PR DESCRIPTION
**Pull request by teodossidossev**
* The base prefix takes its default value from the priority list:
(1) The default namespace of the repository 
(2) The repository’s “Base URL”

* Change the prefix-list by default to be empty and only populated after they are used
* Change the prefix-list label
* Tests and fixtures altered according changes

### CODE REVIEW

**Comment by svilenvelikov** [comment]
Provide PR description please.

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "start": "ng serve",
     "build": "ng build",
     "build:prod": "ng build --prod --base-href=/orefine/extension/mapping-editor/ && mv dist/rdf-mapping-editor/assets/i18n/en.json dist/rdf-mapping-editor/assets/i18n/en.js",
-    "test": "npx jest",
+    "test": "npx jest --passWithNoTests",
```
Comment (svilenvelikov) [inline package.json:10]: Why?

File: src/app/services/rest/mapping-definition.service.ts [hunk 2]
```diff
@@ -7,24 +7,33 @@ import {ActivatedRoute} from '@angular/router';
...
 
-  static createNewMapping() {
+  static createNewMapping(name) {
     const newMapping = EMPTY_MAPPING;
+    newMapping.baseIRI = name;
     newMapping.subjectMappings = [];
     return newMapping;
   }
 
+  static getBaseNamespace(namespaces: {}) {
```
Comment (svilenvelikov) [inline src/app/services/rest/mapping-definition.service.ts:32]: There is a `Namespaces` type in the models package which you can use

File: src/app/services/rest/repository.service.ts [hunk 3]
```diff
@@ -38,6 +44,14 @@ export class RepositoryService {
     }));
   }
 
+  getBaseURL(): Observable<Namespaces> {
+    return this.getGdbApiURL('').pipe(switchMap((fullUrl) => {
+      return this.httpClient.get<any>(fullUrl, {}).pipe(map((res) => {
```
Comment (svilenvelikov) [inline src/app/services/rest/repository.service.ts:49]: I started to hate rxjs and I believe that we write it in very wrong way. IMO such code is barely readable and maintainable :(

File: src/app/services/rest/repository.service.ts [hunk 4]
```diff
@@ -38,6 +44,14 @@ export class RepositoryService {
     }));
   }
 
+  getBaseURL(): Observable<Namespaces> {
+    return this.getGdbApiURL('').pipe(switchMap((fullUrl) => {
```
Comment (svilenvelikov) [inline src/app/services/rest/repository.service.ts:48]: Why do you even need to have observable from the getGdbApiURL? Why is that needed? It's completely syncronous code and you can just make uit return plain string and omit the extra two level nesting here!

**Review by svilenvelikov — DISMISSED** [review]

**Review by desislava-hristova-ontotext — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (teodossidossev) [reply package.json:10]: I removed all useless tests, so no one left. And without this flag the step fails the build

**Comment by teodossidossev** [comment]
Jenkins retry build

**Comment by desislava-hristova-ontotext** [comment]
Jenkins retry build

**Review by desislava-hristova-ontotext — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (svilenvelikov) [reply package.json:10]: And why we keep the step then?
